### ISSUE TITLE
RL policy selects unplayable actions (no legality/affordability check at inference)

### ISSUE THREAD

**Issue by Talor-A** [Talor-A/forge#11]
## Summary

During a 50-game evaluation of the imitation-trained decision model vs. the heuristic AI (model served via `serving/model_server.py`, run with `rltrain evaluate`), the RL policy won **17/50 (34%)**. A recurring failure mode contributing to losses: **the policy ranks and attempts actions it cannot legally or affordably execute.** The chosen action then fails at resolution, wasting the play and leaving the model in a degraded board state.

## Evidence

From a game in the eval run (game 41), the priority head chose to cast a spell it could not pay for:

```
RL_PRIORITY_PLAY: Primal Might (Pump) -> target: Rishkar, Peema Renegade (of 4 legal)
ComputerUtilMana: payManaCost() cost was not paid for Target creature you control gets +X/+X ... (Primal Might). Didn't find what to pay for {1}
[RL_41] AI failed to play Primal Might (87) [Stack]
```

The model picked the highest-scoring candidate from the priority head without verifying the mana cost could actually be paid, so the cast aborted.

## Why this matters

- A game is hundreds of chained decisions; each wasted/illegal action compounds into board disadvantage.
- The heuristic AI does affordability/legality lookahead before committing, so it never burns a turn this way — part of why imitation cloning sits well below the teacher (34% win rate).

## Proposed fix

Add a **legality/affordability mask at inference** so the policy can never select an unexecutable action:
- Before argmax/sampling over candidate logits, mask out candidates the engine reports as unplayable (insufficient mana, no legal target, etc.).
- The candidate legality information is already available engine-side at decision time; the masking can live in `PlayerControllerRL` / the decision service, or in the model server response handling.

This is the cheapest of the three improvement paths (vs. RL fine-tuning or MCTS-at-inference) and directly removes the wasted-action class of bug.

## Repro

```
python serving/model_server.py --model <decisions checkpoint> --port 50051 --device cpu
# then, in forge-gui-desktop:
java ... rltrain evaluate -D <decks> -n 50 -t 4 --host localhost --port 50051 -o /tmp/eval
```

Win rate ~34% (n=50, ±~13% CI), 0 server errors.
